components/external-services-mock/internal/systemfetcher/handler.go

```diff
@@ -60,10 +61,24 @@ func (s *SystemFetcherHandler) HandleFunc(rw http.ResponseWriter, req *http.Requ
 		}
 		return
 	}
+
+	resp := []byte("[]")
+	if len(s.mockedSystems) > 0 {
+		resp = s.mockedSystems[0]
+		s.mockedSystems = s.mockedSystems[1:]
+	}
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
-	_, err := rw.Write(s.mockedSystems)
+	_, err := rw.Write(resp)
 	if err != nil {
 		httphelpers.WriteError(rw, errors.Wrap(err, "error while writing response"), http.StatusInternalServerError)
 	}
 }
+
+func (s *SystemFetcherHandler) HandleReset(rw http.ResponseWriter, _ *http.Request) {
```

Comment:
I'm also wondering for the purpose of this reset handler.

Reply:
When an e2e test is done we want to make sure that no leftover mock responses are left which can potentially break the next one. That is why all system-fetcher tests [have a deferred cleanup](https://github.com/kyma-incubator/compass/pull/1985/files#diff-8ab34488d3ef47b9a7de83c838c9fab56baaa87ef7b2d022de34fc46318e61d3R230) for the systems that they have mocked to the external-services-mock. This handler simply clears the queue and gets called at the end of each test.